src/v/utils/rjson_allocator.h

```diff
@@ -0,0 +1,26 @@
+#include <rapidjson/allocators.h>
+#include <rapidjson/document.h>
+#include <rapidjson/encodings.h>
+
+namespace rjson_allocator {
+class MyAllocator : public rapidjson::CrtAllocator {
+public:
+    void* Malloc(size_t size) {
+        if (!size) {
+            return NULL;
+        }
+
+        void* res = rapidjson::CrtAllocator::Malloc(size);
+        if (NULL == res) {
+            throw std::bad_alloc{};
+        }
+
+        return res;
+    }
+};
+
+using Document = rapidjson::GenericDocument<
```

Comment:
This probably doesn't belong in a file named "allocator".

Are there any other types where the allocator should be replaced?

Reply:
moved it to the `document.h`

for other types I think we can do it in the separate PR